src/api/dashboard.js

```diff
@@ -56,14 +63,29 @@ Dashboard.prototype = {
     this._dashboard.vis.mapvis.map.setBounds([state.map.ne, state.map.sw]);
   },
 
-  onStateChanged: function (callback, shareURLs) {
-    this._dashboard.vis.once('dataviewsFetched', function () {
-      this._dashboard.widgets._widgetsCollection.initialState();
-      this._dashboard.widgets._widgetsCollection.each(function (m) {
-        m.applyInitialState();
-      }, this);
-      shareURLs === true && this._bindChange(callback);
-    }, this);
+  _widgetsAlreadyInitialized: function () {
+    var widgetsCollection = this._dashboard.widgets._widgetsCollection;
```

Comment:
I have added this last thing because testing it in Builder I noticed that when there are no widgets added, callback was not triggered. Thanks acceptance you exist :joy:.